rendermodules/pointmatch/swap_point_match_collections.py

```diff
@@ -39,51 +116,22 @@ def run(self):
-                                                                            groupid, 
-                                                                            owner=self.args['match_owner'],
-                                                                            render=self.render)
-                targetmatches = renderapi.pointmatch.get_matches_with_group(self.args['target_collection'],
-                                                                            groupid,
-                                                                            owner=self.args['match_owner'],
-                                                                            render=self.render)
-                
-                # get the qgroupIds in source matches
-                sourcegroupIds = [m['qGroupId'] for m in sourcematches]
-
-                # delete the matches between groupid and sourcegroupIds
-                for sg in sourcegroupIds:
-                    renderapi.pointmatch.delete_point_matches_between_groups(self.args['source_collection'],
-                                                                             groupid,
-                                                                             sg,
-                                                                             owner=self.args['match_owner'],
-                                                                             render=self.render)
-                targetgroupIds = [m['qGroupId'] for m in targetmatches]
-                for tg in targetgroupIds:
-                    renderapi.pointmatch.delete_point_matches_between_groups(self.args['target_collection'],
-                                                                             groupid,
-                                                                             tg,
-                                                                             owner=self.args['match_owner'],
-                                                                             render=self.render)
-                
-                # import source matches into target collection
-                renderapi.pointmatch.import_matches(self.args['target_collection'],
-                                                    sourcematches,
-                                                    owner=self.args['match_owner'],
-                                                    render=self.render)
-                # import target matches into source collection
-                renderapi.pointmatch.import_matches(self.args['source_collection'],
-                                                    targetmatches,
-                                                    owner=self.args['match_owner'],
-                                                    render=self.render)
-                zvalues.append(z)
+        zvalues = [z for z, n in zip(ids, output_bool) if output_bool ]    
```

Comment:
This should test for n:
```suggestion
        zvalues = [z for z, n in zip(ids, output_bool) if n]    
```